azure-mgmt-trafficmanager/src/main/java/com/microsoft/azure/management/trafficmanager/TrafficManagerEndpoint.java

```diff
@@ -0,0 +1,500 @@
+             * @return the next stage of the endpoint definition
+             */
+            WithAttach<ParentT> withRoutingWeight(int weight);
+        }
+
+        /**
+         * The stage of the traffic manager endpoint definition allowing to specify the endpoint priority.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithRoutingPriority<ParentT> {
+            /**
+             * Specifies the priority for the endpoint that will be used when the parent profile is configured with
+             * Priority routing method {@link TrafficRoutingMethod#PRIORITY}.
+             *
+             * @param priority the endpoint priority
+             * @return the next stage of the endpoint definition
+             */
+            WithAttach<ParentT>  withRoutingPriority(int priority);
+        }
+
+        /**
+         * The stage of the traffic manager endpoint definition allowing to disable the endpoint.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithTrafficDisabled<ParentT> {
+            /**
+             * Specify that this endpoint should be excluded from receiving traffic.
+             *
+             * @return the next stage of the endpoint definition
+             */
+            WithAttach<ParentT> withTrafficDisabled();
+        }
+
+        /** The final stage of the traffic manager profile endpoint definition.
+         * <p>
+         * At this stage, any remaining optional settings can be specified, or the traffic manager profile endpoint
+         * definition can be attached to the parent traffic manager profile definition using {@link TrafficManagerEndpoint.DefinitionStages.WithAttach#attach()}.
+         * @param <ParentT> the return type of {@link TrafficManagerEndpoint.DefinitionStages.WithAttach#attach()}
+         */
+        interface WithAttach<ParentT> extends
+                Attachable.InDefinition<ParentT>,
+                DefinitionStages.WithRoutingWeight<ParentT>,
+                DefinitionStages.WithRoutingPriority<ParentT>,
+                DefinitionStages.WithTrafficDisabled<ParentT> {
+        }
+    }
+
+    /**
+     * The entirety of a traffic manager profile endpoint definition as a part of parent update.
+     *
+     * @param <ParentT> the return type of the final {@link Attachable#attach()}
+     */
+    interface UpdateDefinition<ParentT> extends
+            UpdateDefinitionStages.Blank<ParentT>,
+            UpdateDefinitionStages.WithEndpointType<ParentT>,
+            UpdateDefinitionStages.WithEndpointThreshold<ParentT>,
+            UpdateDefinitionStages.WithSourceTrafficLocation<ParentT>,
+            UpdateDefinitionStages.WithAttach<ParentT> {
+    }
+
+    /**
+     * Grouping of traffic manager profile endpoint definition stages as a part of parent traffic manager
+     * profile update.
+     */
+    interface UpdateDefinitionStages {
```

Comment:
just a note to address the corresponding comments in the UpdateDefinition, since it's probably identical to Definition